FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc

```diff
@@ -0,0 +1,91 @@
+#include "FastSimulation/CTPPSFastGeometry/interface/CTPPSToFDetector.h"
+#include <math.h>
+
+
+CTPPSToFDetector::CTPPSToFDetector(int ncellx,int ncelly, std::vector<double>& cellw,double cellh,double pitchx,double pitchy,double pos, int res):
+    NCellX(ncellx),NCellY(ncelly),CellW(cellw),CellH(cellh),PitchX(pitchx),PitchY(pitchy),fToFResolution(res),DetPosition(pos) {
+        // the vertical positions starts from the negative(bottom) to the positive(top) corner
+        // vector index points to the row number from below
+        CellRow.push_back(std::pair<double,double>(-CellH/2.,CellH/2.));
+        // vector index points to the column number
+        for(int i=0;i<NCellX;i++) {
+            double x1 = 0., x2 = 0.; 
+            if(i==0) {DetW = PitchX; x1 = -(DetPosition+DetW);}
+            else x1 = -DetPosition+DetW; //DetPosition - shift the limit of a column depending on the detector position
+            x2 = x1-CellW.at(i);
+            DetW += (x2-x1)-PitchX; 
+            CellColumn.push_back(std::pair<double,double>(x1,x2));
```

Comment:
add CellColumn.reserve(NCellX) before the loop (and for similar constructs below)